tracee/pipeline.go

```diff
@@ -0,0 +1,250 @@
+package tracee
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"sync"
+)
+
+func (t *Tracee) runEventPipeline(done <-chan struct{}) error {
+	var errcList []<-chan error
+
+	// Source pipeline stage.
+	dataBuffChan, errc, err := t.readEventFromChannel(done)
```

Comment:
I think this step is redundant. you already have the events to be consumed in the `eventsChannel` what does `readEventFromChannel` add on top of that?